src/main/java/uk/gov/justice/hmpps/prison/service/BookingService.java

```diff
@@ -325,36 +326,35 @@ public List<ScheduledEvent> getBookingActivities(final Long bookingId, final Loc
     @Transactional
     @PreAuthorize("hasRole('ROLE_PAY')")
     public void updateAttendance(final String offenderNo, final Long activityId, @Valid @AttendanceTypesValid final UpdateAttendance updateAttendance) {
-        updateAttendance(activityId, updateAttendance, getLatestBookingByOffenderNo(offenderNo));
+        // Copy flags from the PAYABLE_ATTENDANCE_OUTCOME reference table
+        final var activityOutcome = bookingRepository.getPayableAttendanceOutcome("PRISON_ACT", updateAttendance.getEventOutcome());
+        updateAttendance(activityId, updateAttendance, getLatestBookingByOffenderNo(offenderNo).getBookingId(), activityOutcome);
     }
 
     @Transactional
     @PreAuthorize("hasRole('ROLE_PAY')")
     public void updateAttendance(final Long bookingId, final Long activityId, @Valid @AttendanceTypesValid final UpdateAttendance updateAttendance) {
-        updateAttendance(activityId, updateAttendance, getLatestBookingByBookingId(bookingId));
+        final var activityOutcome = bookingRepository.getPayableAttendanceOutcome("PRISON_ACT", updateAttendance.getEventOutcome());
+        updateAttendance(activityId, updateAttendance, getLatestBookingByBookingId(bookingId).getBookingId(), activityOutcome);
     }
 
     @Transactional
     @PreAuthorize("hasRole('ROLE_PAY')")
     public void updateAttendanceForMultipleBookingIds(final Set<BookingActivity> bookingActivities, @Valid @AttendanceTypesValid final UpdateAttendance updateAttendance) {
-        bookingActivities.forEach(bookingActivity -> updateAttendance(bookingActivity.getActivityId(), updateAttendance, getLatestBookingByBookingId(bookingActivity.getBookingId())));
-    }
-
-    private void updateAttendance(final Long activityId, final UpdateAttendance updateAttendance, final OffenderSummary offenderSummary) {
-        validateActivity(activityId);
```

Comment:
It turned out that both calling getLatestBookingByBookingId(), and validateActivity(), were redundant as the same checks are done in the update itself